client/components/button-group/index.jsx

```diff
@@ -1,36 +1,21 @@
 /**
  * External dependencies
  */
-
-import React, { PureComponent } from 'react';
+import React from 'react';
 import classNames from 'classnames';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-class ButtonGroup extends PureComponent {
-	static propTypes = {
-		children( props ) {
-			let error = null;
-			React.Children.forEach( props.children, ( child ) => {
-				if ( child && ( ! child.props || child.props.type !== 'button' ) ) {
-					error = new Error( 'All children elements should be a Button.' );
-				}
-			} );
-			return error;
-		},
-	};
-
-	render() {
-		const buttonGroupClasses = classNames( 'button-group', this.props.className, {
-			'is-busy': this.props.busy,
-			'is-primary': this.props.primary,
-		} );
+const ButtonGroup = ( { busy, children, className, primary } ) => {
+	const buttonGroupClasses = classNames( 'button-group', className, {
+		'is-busy': busy,
+		'is-primary': primary,
+	} );
 
-		return <span className={ buttonGroupClasses }>{ this.props.children }</span>;
-	}
-}
+	return <span className={ buttonGroupClasses }>{ children }</span>;
+};
 
 export default ButtonGroup;
```

Comment:
As the original component extended `PureComponent` we should wrap this in `React.memo` to maintain the same behavior.

Reply:
Good call - I've addressed that in https://github.com/Automattic/wp-calypso/pull/50984/commits/a12664d066e4bca0ce1ca375dd9f5ece84d449e4. Also, removed an obsolete test that was expecting the old behavior was still there in https://github.com/Automattic/wp-calypso/pull/50984/commits/c1c47f5544971e5cc7dcaca08af30d6328d8699f.